### PR TITLE
HLRC: Add activate watch action

### PR DESCRIPTION
**Pull request by iverase**
Adds activate watch action to the high level rest client.

Relates #29827

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java
Comment (pcsanwald) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java]: do we need to override `toString` here? seems relatively harmless, but wasn't sure if it's necessary.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherIT.java
Comment (pcsanwald) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherIT.java]: I'd split this into a separate test case, `testActivateWatchThatExists` and `testActivateWatchThatDoesNotExist`

File: client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherRequestConvertersTests.java
Comment (pcsanwald) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherRequestConvertersTests.java]: I think `testActivateWatchRequestConversion` is a clearer name.

**Review by pcsanwald — APPROVED** [review]
A few small naming suggestions and one test suggestion, but I don't want to block merging on re-review.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherIT.java
Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/WatcherIT.java]: +1

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java]: We will have a new ay to deal with the docs+test, we should wait to merge until it is merged.

**Review by hub-cap — CHANGES_REQUESTED** [review]
Wow im super sorry, i had done a review and forgot to click submit :/

Update to today, and we will very soon have a better way to test responses (you do not have a response test) and to generate docs, in the next day or two. We should hold off until then to merge. Feel free to fix up my nits tho.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java
Comment (iverase) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java]: @hub-cap which is the issue that it is introducing this change?

Comment (hub-cap) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/WatcherDocumentationIT.java]: #34125 and it looks like its merged!

**Comment by iverase** [comment]
@hub-cap Added response test and changed the docs. I think it is getting closer.

**Comment by hub-cap** [comment]
Yea we backport all of the HLRC APIs to the current 6.x. Some of them landed in 6.4, most will land in 6.5

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java]: Any reason this is Closeable?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java]: there is a `Objects.requireNonNull` helper you can use here

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchResponse.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchResponse.java]: This is typically on teh top of these classes.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchResponse.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchResponse.java]: this too is on top

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java
Comment (hub-cap) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ActivateWatchRequest.java]: You may also have a look at `TimedRequest`, assuming this needs the timeouts (I dont know if it does or not, you would have to research the transport layer action)

**Review by hub-cap — APPROVED** [review]
minor nits, then :shipit:

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by iverase** [comment]
@elasticmachine test this please

**Comment by iverase** [comment]
@hub-cap The commit cannot be back ported as WatchStatus has not been back ported. If necessary I can coordinate with @jtibshirani to do so.

**Comment by jtibshirani** [comment]
Just backported it to 6.x in 83e30eeeb5a5c255bfd32f40faf60c62ca75e69b, sorry for the delay!

**Pull request merged**
